lib/sc_client.rb

```diff
@@ -13,6 +15,9 @@
 class ScClient
   include Emitter
   include DataModels
+  include Reconnect
+
+  attr_accessor :reconnect_interval, :max_reconnect_interval, :reconnect_decay, :max_attempts, :attempts_made
```

Comment:
Don't need to add attr_accessor for variables used in the class itself and unless we are using strong params.

Reply:
* attr_accessors are used here to avoid unnecessary for getter-setter methods, strong params are not applicable here